stub_uploader/upload.py

```diff
@@ -25,24 +25,43 @@ def upload(
     print("Building and uploading stubs for:", ", ".join(distributions))
     print()
     for distribution in to_upload:
-        print(f"Building stubs for {distribution}... ", end="")
-        metadata = read_metadata(typeshed_dir, distribution)
-        recursive_verify(metadata, typeshed_dir)
+        upload_distribution(typeshed_dir, distribution, commit, dry_run=dry_run)
 
+
+def upload_distribution(
+    typeshed_dir: str, distribution: str, commit: str | None, *, dry_run: bool
+) -> None:
+    already_uploaded = False
+
+    print(f"Building stubs for {distribution}... ", end="")
+    metadata = read_metadata(typeshed_dir, distribution)
+    recursive_verify(metadata, typeshed_dir)
+
+    try:
         version = determine_stub_version(metadata)
-        if commit:
-            update_changelog(
-                typeshed_dir, commit, distribution, version, dry_run=dry_run
-            )
-
-        temp_dir = build_wheel.main(typeshed_dir, distribution, version)
-        print(f"ok, version {version}")
-
-        print(f"Uploading stubs for {distribution}... ", end="")
-        if dry_run or not metadata.upload:
-            print(f"skipped")
-        else:
-            subprocess.run(["twine", "upload", os.path.join(temp_dir, "*")], check=True)
-            uploaded_packages.add(distribution)
-            print(f"ok")
-        print()
+    except AlreadyUploadedError as e:
+        version = str(e.version)
+        already_uploaded = True
```

Comment:
For readability, I would do this rather than setting it to False on line 34. This way it's clearer that the try/except here is the sole determiner of whether this value is True or False 

```suggestion
        already_uploaded = True
    else:
        already uploaded = False
```